Fix MSVC build: remove GCC-only 'd' suffix on double literals

PozzettiAndrea opened the pull request:
The 'd' suffix on double literals (e.g., 1e-6d, 0.0d) is a GCC extension that MSVC does not support. Use standard C++ double literals instead.

File changed: src/convert/flexible_dual_grid.cpp

microsoft-github-operations[bot] committed:
Initial commit

JeffreyXiang committed:
Init Release

JeffreyXiang committed:
Add demo link

JeffreyXiang committed:
Fix website url

JeffreyXiang committed:
update paper link

JeffreyXiang committed:
update texturing pipeline

Deathdadev commented:
While these changes (including #61, #60, #59) correctly resolve issues for Windows users, **they introduce breaking changes for Linux environments**.

These PRs port Linux syntax specifically to Windows, similar to approaches taken in other forks (See #4 for information, [Deathdadev/TRELLIS.2-Windows](https://github.com/Deathdadev/TRELLIS.2-Windows) (my own fork), [visualbruno/ComfyUI-Trellis2](https://github.com/visualbruno/ComfyUI-Trellis2)).

Since this repository is tested and designed for Linux, we cannot simply replace the syntax. Instead, we need to implement cross-platform logic to ensure the code runs correctly on both systems.

Edit:
Apologies, #61 is unrelated to Linux syntax

PozzettiAndrea commented:
@Deathdadev I'm all for cross-platform logic! :)

I don't think my changes introduce any breaking changes for Linux though:
  - PR #59: The d suffix on double literals is a GCC extension, not standard C++. 1e-6 and 0.0 are already doubles by default in standard C++, and the code should then compile correctly on GCC, Clang, and MSVC.
  - PR #60: The explicit (int64_t) casts are standard C++ and work identically on all compilers. They just make implicit conversions explicit.
  - PR #61: This is pure Python code (not platform specific)

These changes move the codebase toward standard C++, to my understanding they don't use "Windows syntax"
Please let me know if you see any specific issue!

Deathdadev commented:
@PozzettiAndrea Sorry about that, I almost completely forgot about standard C++ altogether! I did some trial and error just now in case there was something I could pick up on, and you are correct.

Thanks for clarifying!

PozzettiAndrea committed:
Fix MSVC compilation: remove non-standard 'd' suffix on double literals

The 'd' suffix on double literals (e.g., 1e-6d, 0.0d) is a GCC extension
that MSVC does not support. Use standard C++ double literals instead.

File changed: src/convert/flexible_dual_grid.cpp
